### DISCUSSION TITLE
Using `tree-sitter parse` with a pre-built .so file?

### DISCUSSION THREAD

**Discussion by osa1** [tree-sitter/tree-sitter#812] [Q&A] [answered]
My parser requires a lexer implemented a different language. I have a build script that generates the final `.so` file that includes the lexer, but I can't use it with `tree-sitter parse` to quickly test my changes. I checked `tree-sitter parse --help` but it seems like `parse` command doesn't support specifying the `.so` file path, right? Is there a way to run `parse` with a given `.so` file for the parser?

**Comment by osa1** [comment, marked as answer]
I couldn't find how to do this so I quickly implemented a tool like `tree-sitter-cli` that loads the parser from a given .so file: https://github.com/osa1/tree-sitter-parse

**Reply by maxbrunsfeld** [reply]
Nice work. If you can think of an intuitive way that this functionality could be incorporated into the main CLI's `tree-sitter parse` command, I would be open to adding that feature.

Maybe just a new CLI flag like `--language`? Or we could overload the `--scope` flag so that if you pass an absolute path to a `.so`/`.dll`/`.dylib` file, it will use that.
